### ISSUE TITLE
Can no longer group interface members

### ISSUE THREAD

**Issue by ejain** [mvdan/gofumpt#186]
The new "don't group interface members of different kinds" rule in 0.2.1 also appears to prohibit the following grouping:

```go
type i1 interface {
	DoA()
	DoB()
	
	UndoA()
	UndoB()
}```

Was that intentional?

**Comment by mvdan** [comment]
See the discussion in https://github.com/mvdan/gofumpt/issues/85. The premise is that this kind of grouping is very common in structs, but not nearly as common in interfaces. I think the vast majority of interfaces fall in one of two categories:

1) They are intended for external use, and each method is documented. gofumpt will not remove newlines between interface methods if either method has a godoc comment.
2) They do not have any comments, they are "flat", akin to a multi-line slice or map literal.

So my suggestion would be that, if you want to keep a logical separation between methods, you could document the methods themselves, or you could add a separation comment to clarify the grouping - then gofumpt shouldn't join the lines:

```
type i1 interface {
	DoA()
	DoB()

	// Undo methods below this point.
	
	UndoA()
	UndoB()
}
```

Maybe this rationale is flawed, I'm not sure. I'm open to ideas. cc @flimzy @Zamiell @ainar-g @Oiyoo

**Comment by mvdan** [comment]
I slightly lean towards removing the feature entirely; its value add feels rather small, and it seems relatively prone to fighting the user when they are using empty lines as logical separators in reasonable ways.

We don't remove empty lines inside slice or map literals, for example, and I don't think we should start doing so either.

**Comment by ainar-g** [comment]
@mvdan, in my projects, I've basically moved to documenting every exported and unexported method.  That was for reasons unrelated to `gofumpt`, but it does give me the additional freedom to (not) group the methods however I want, so the removal of the feature wouldn't bother me too much.

**Comment by flimzy** [comment]
I think #85 was known guess work. If we've proven our guess about grouping interfaces was wrong, it's not unreasonable to revert that change.

**Comment by mvdan** [comment]
I'll let this sit for a week, in case there are any objections. If there are none, I think I'll remove the feature. Thanks all for the input so far :)

**Comment by jaapz** [comment]
Thanks for fixing this! Is it going to be released as 0.2.2?

**Comment by mvdan** [comment]
I intend to release v0.3.0 by tomorrow, to be included in gopls v0.8.0 hopefully.
